packages/core/src/__integrationtests__/PublicCredentials.spec.ts

```diff
@@ -0,0 +1,315 @@
+      attester.uri
+    )
+
+    const publicCredentialEntry =
+      await api.call.publicCredentials.getCredential(credentialId)
+    expect(publicCredentialEntry.isSome).toBe(true)
+
+    const completeCredential = await PublicCredential.credentialFromChain(
+      credentialId,
+      publicCredentialEntry
+    )
+
+    // Verify that the retrieved credential matches the input one, plus the generated ID and the attester DID.
+    expect(completeCredential).toEqual(
+      expect.objectContaining({
+        ...latestCredential,
+        id: credentialId,
+        attester: attester.uri,
+        revoked: false,
+      })
+    )
+  })
+
+  it('should be possible to issue a second credential to the same asset and retrieve both of them', async () => {
+    latestCredential = {
+      claims: {
+        name: `Certified NFT collection with id ${UUID.generate()}`,
+      },
+      cTypeHash: CType.getHashForSchema(nftNameCType),
+      delegationId: null,
+      subject: assetId,
+    }
+    const encodedPublicCredential = PublicCredential.toChain(latestCredential)
+    const storeTx = api.tx.publicCredentials.add(encodedPublicCredential)
+    const authorizedStoreTx = await Did.authorizeTx(
+      attester.uri,
+      storeTx,
+      attesterKey.getSignCallback(attester),
+      tokenHolder.address
+    )
+    await submitTx(authorizedStoreTx, tokenHolder)
+
+    const encodedAssetCredentials =
+      await api.call.publicCredentials.getCredentials(assetId, null)
+    const assetCredentials = await PublicCredential.credentialsFromChain(
+      encodedAssetCredentials
+    )
+
+    // We only check that we return two credentials back.
+    // We don't check the content of each credential.
+    expect(assetCredentials).toHaveLength(2)
+  })
+
+  it('should be possible to retrieve 100 credentials for the same asset', async () => {
+    // Issue 98 more credentials
+    const credentialCreationTxs = [...Array(98)].map(() => {
+      latestCredential = {
+        claims: {
+          name: `Certified NFT collection with id ${UUID.generate()}`,
+        },
+        cTypeHash: CType.getHashForSchema(nftNameCType),
+        delegationId: null,
+        subject: assetId,
+      }
+      const encodedPublicCredential = PublicCredential.toChain(latestCredential)
+      return api.tx.publicCredentials.add(encodedPublicCredential)
+    })
+    const authorizedBatch = await Did.authorizeBatch({
+      batchFunction: api.tx.utility.batchAll,
+      did: attester.uri,
+      extrinsics: credentialCreationTxs,
+      sign: attesterKey.getSignCallback(attester),
+      submitter: tokenHolder.address,
+    })
+    await submitTx(authorizedBatch, tokenHolder)
+
+    const encodedAssetCredentials =
+      await api.call.publicCredentials.getCredentials(assetId, null)
+    const assetCredentials = await PublicCredential.credentialsFromChain(
+      encodedAssetCredentials
+    )
+
+    // We only check that we return all the twenty credentials back.
```

Comment:
This is the reason I prefer self-documenting code to comments. It’s very easy to forget to update the comment when you’re updating the code. It’s harder to forget to update the code.